pkg/models/user.go

```diff
@@ -227,6 +228,7 @@ type UserSearchHitDTO struct {
 	IsDisabled    bool      `json:"isDisabled"`
 	LastSeenAt    time.Time `json:"lastSeenAt"`
 	LastSeenAtAge string    `json:"lastSeenAtAge"`
+	AuthModule    string    `json:"authModule"`
```

Comment:
AuthModule will have to be a list of strings as a user can be linked with more than one module.

Reply:
Hm, @xlson could you point me to the kind of example - I can't find that `GetAuthInfo()` can return multiple modules. I'm looking into the tests and it returns the last module only:
https://github.com/grafana/grafana/blob/fda7e686f34e91eae96dbc1e44d178297a409066/pkg/services/sqlstore/user_auth_test.go#L173

Reply:
After discussing this with @torkelo we think it would be a good idea to have the api return an array (for correctness if nothing else, not adding this information could cause problems in the future). But in the ui it might be fine to only show he one used most recently.

Reply:
@torkelo could you explain a little bit why do we need to have multiple auth modules in `/api/users/search` endpoint? Getting one module is quite simple - just a left join with `user_auth` table. But returning all auth modules requires a more complex method. So I'm trying to beware of adding this without a strong reason. Another problem in this case is to determine which module is active. If we return a list of modules, user should rely on modules order which is not a robust solution I think. I can extend `/api/users/:id` endpoint and return all auth modules there (and show all modules on the user's profile page, for instance). So what's the main reason to have all modules in `/api/users/search`?
cc @xlson

Reply:
I think you misunderstood the comment, the truth is that it's an array (multiple), but we can use the simple left join (single) for now, but lets have the API be an array so we can correct it in the future and return all auth modules without making a breaking api change.

Reply:
Guess, the same changes should be in `UserProfileDTO` struct?
```go
type UserProfileDTO struct {
	Id             int64  `json:"id"`
	Email          string `json:"email"`
	Name           string `json:"name"`
	Login          string `json:"login"`
	Theme          string `json:"theme"`
	OrgId          int64  `json:"orgId"`
	IsGrafanaAdmin bool   `json:"isGrafanaAdmin"`
	IsDisabled     bool   `json:"isDisabled"`
	AuthModule     string `json:"authModule"`
}
```